dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -5702,6 +5567,13 @@ bool DynamicDataImpl::DataContainer::serialize_sequence_value(DCPS::Serializer&
 bool DynamicDataImpl::DataContainer::get_index_to_id_map(IndexToIdMap& index_to_id,
                                                          CORBA::ULong bound) const
 {
+  for (const_single_iterator it = single_map_.begin(); it != single_map_.end(); ++it) {
+    CORBA::ULong index;
+    if (!data_->get_index_from_id(it->first, index, bound)) {
+      return false;
+    }
+    index_to_id[index] = it->first;
+  }
```

Comment:
This came up in the course of trying to get the xcdr test to work with DynamicData. The specific case was serialization of a sequence of strings was causing the strings to be empty because they didn't check `single_map_`.

I'm don't think I did anything to change the usage. This is where it's getting called in the string sequence case:

https://github.com/OpenDDS/OpenDDS/blob/79c7a52b4ac6f1abed50ea67cf1f1ee321ae1618/dds/DCPS/XTypes/DynamicDataImpl.cpp#L5003-L5010

I don't feel confidant in what the comment should say, For now I'm going to change it to "Helper function for serializing sequences and arrays". If you have more to add to that I won't be opposed.

Reply:
I just checked the code and you're right. I think the usage of `get_index_to_id_map` was changed during development but I forgot to update the function itself.